tests/e2e/vmgroup_volumeops_test.go

```diff
@@ -0,0 +1,300 @@
+)
+
+const (
+	volName1                 = "vmgroup_test_vol_1"
+	volName2                 = "vmgroup_test_vol_2"
+	defaultVg                = "_DEFAULT"
+	defCname                 = "def_test_ctr"
+	defVgName                = "def_test_vg"
+)
+
+// Tests to validate behavior with the __DEFAULT_ vmgroup.
+
+var vmHosts = []string{os.Getenv("VM1"), os.Getenv("VM2")}
+var host, vmgroup, ds1, ds2, cname string
+
+func getEnv(t *testing.T) {
+	host = os.Getenv("ESX")
+	vmgroup = os.Getenv("T1")
+	ds1 = os.Getenv("DS1")
+	ds2 = os.Getenv("DS2")
+	cname = os.Getenv("CNAME")
+
+	if vmgroup == "" {
+		vmgroup = defVgName
+	}
+
+	if ds1 == "" || ds2 == "" {
+		t.Errorf("Unknown datastores, please set DS1 and DS2 to valid datastore names in the environment")
+	}
+
+	if cname == "" {
+		cname = defCname
+	}
+}
+
+func createVolumeOnDefaultVg(t *testing.T, name string) {
+	// 1. Create the volume on each host
+	for _, vm := range vmHosts {
+		_, err := dockercli.CreateVolume(vm, name, "")
+		if err != nil {
+			t.Errorf("Error while creating volume - %s on VM - %s, err - %s", name, vm, err.Error())
+		}
+	}
+
+	// 2. Verify the volume is created on the default vm group
+	cmd := admincli.ListVolumes + " | grep " + defaultVg + " | grep " + name
+	val, err := ssh.ExecCmd(host, cmd)
+	if err != nil {
+		t.Errorf("Error while listing volumes [%s] in default vmgroup on host %s [%s] err - %s", cmd, host, err.Error())
+	}
+	if val == "" {
+		t.Errorf("Volume %s not found in default vmgroup", name)
+	}
+}
+
+// TestVolumeCreateOnDefaultVg - Verify that volumes can be created on the
+// default volume group with default permissions, then attached and deleted
+func TestVolumeCreateAndAttachOnDefaultVg(t *testing.T) {
+	getEnv(t)
+	// Create a volume in the default group
+	createVolumeOnDefaultVg(t, volName1)
+
+	// 1. Verify volume can be mounted and used on at least one of the VMs
+	_, err := dockercli.AttachVolume(vmHosts[0], volName1, cname)
+	if err != nil {
+		t.Errorf("Error while attaching volume - %s on VM - %s, err - %s", volName1, vmHosts[0], err.Error())
+	}
+
+	// 2. Delete the volume in the default group
+	_, err = dockercli.DeleteVolume(vmHosts[0], volName1)
+	if err != nil {
+		t.Errorf("Error while deleting volume - %s in the default vmgroup, err - %s", volName1, err.Error())
+	}
+	t.Logf("Passed - Volume create and attach on default vmgroup")
+}
+
+// TestVolumeAccessAcrossVolumeGroups - Verify volumes can be accessed only
+// from VMs that belong to the volume group
+func TestVolumeAccessAcrossVolumeGroups(t *testing.T) {
+	getEnv(t)
+	// Create a volume in the default group
+	createVolumeOnDefaultVg(t, volName1)
+
+	// 1. Create a new vmgroup, with ds2 as the datastore
+	cmd := admincli.CreateVolumeGroup + vmgroup + " --vm-list " + vmHosts[0] + " --default-datastore " + ds2
+	_, err := ssh.ExecCmd(host, cmd)
+	if err != nil {
+		t.Errorf("Error while creating volume group - [%s] err - %s", cmd, err.Error())
+	}
+
```

Comment:
It would be better if we could also verify the new vmgroup created using admincli vmgroup ls.